pkg/scripts/environment.go

```diff
@@ -21,14 +21,19 @@ type Environment struct {
 // GetEnvironment gets the .sti/environment file located in the sources and
 // parse it into []environment
 func GetEnvironment(config *api.Config) ([]Environment, error) {
-	envPath := filepath.Join(config.WorkingDir, api.Source, ".sti", api.Environment)
+	envPath := filepath.Join(config.WorkingDir, api.Source, ".s2i", api.Environment)
 	if _, err := os.Stat(envPath); os.IsNotExist(err) {
-		return nil, errors.New("no environment file found in application sources")
+		// TODO: Remove this when the '.sti/environment' is deprecated.
+		envPath = filepath.Join(config.WorkingDir, api.Source, ".sti", api.Environment)
+		if _, err := os.Stat(envPath); os.IsNotExist(err) {
+			return nil, errors.New("no environment file found in application sources")
+		}
+		glog.Infof("DEPRECATED: Use .s2i/environment instead of .sti/environment")
 	}
 
 	f, err := os.Open(envPath)
 	if err != nil {
-		return nil, errors.New("unable to read .sti/environment file")
+		return nil, errors.New("unable to read custom environment file")
```

Comment:
Yeah i'm aware but if there's an error reading it, the full path is useful.
This should be very rare anyway.

Ben Parees | OpenShift
On Feb 1, 2016 11:57 AM, "Michal Fojtik" <notifications@github.com> wrote:

> In pkg/scripts/environment.go
> <https://github.com/openshift/source-to-image/pull/405#discussion_r51445494>
> :
>
> >  	}
> >
> >  	f, err := os.Open(envPath)
> >  	if err != nil {
> > -		return nil, errors.New("unable to read .sti/environment file")
> > +		return nil, errors.New("unable to read custom environment file")
>
> @bparees <https://github.com/bparees> problem with that is that envPath
> resolves to full path (including the workdir and it looks *really* ugly
> in console then ;-)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/source-to-image/pull/405/files#r51445494>.
>